docs/public-networks/how-to/bonsai-limit-trie-logs.md

```diff
@@ -0,0 +1,207 @@
+2024-02-02 05:46:10.446+00:00 | main | INFO  | TrieLogSubCommand | Finished pruning. Re-estimating trie logs size...
+2024-02-02 05:46:11.023+00:00 | main | INFO  | TrieLogSubCommand | Estimated trie logs size after pruning: 0 B (0 B estimate is normal when using default settings)
+2024-02-02 05:46:11.023+00:00 | main | INFO  | TrieLogSubCommand | Prune ran successfully. We estimate you freed up 9 GiB!
+Prune ran successfully. We estimate you freed up 9 GiB!
+```
+
+If you are using a TOML configuration file, you can run a command similar to the following:
+
+```bash
+sudo /usr/local/bin/besu/bin/besu --config-file=besu-config.toml storage x-trie-log prune
+```
+## Troubleshoot
+
+Troubleshoot common errors that can occur when attempting to use the trie log prune command to reduce your database size for Bonsai Tries. 
+To minimize the occurence of errors, ensure your command specifies the following:
+
+- `--data-storage-format`
+- `--data-path`
+- `--sync-mode`
+
+### Prune command for mainnet
+
+The prune command should look similar to the following:
+
+```bash
+sudo /usr/local/bin/besu/bin/besu --data-path=/var/lib/besu --data-storage-format=BONSAI --sync-mode=SNAP storage x-trie-log prune
+```
+
+Ensure you stop Besu before running the command.
+
+###  Subcommand not working
+
+`java.lang.IllegalArgumentException: Subcommand only works with data-storage-format=BONSAI`
+
+The `--data-storage-format=BONSAI` may be missing. 
+To resolve, add the storage format. 
+The command should look similar to the following:
+
+```bash title="Specify data storage format"
+sudo /usr/local/bin/besu/bin/besu --data-storage-format=BONSAI --data-path=/var/lib/besu --sync-mode=SNAP storage x-trie-log prune
+```
+
+### Column handle not found for segment `TRIE_BRANCH_STORAGE`
+
+`java.lang.RuntimeException: Column handle not found for segment TRIE_BRANCH_STORAGE`
+
+Ensure you specify the `data-path`. 
+Your command should look similar to the following:
+
+```bash
+sudo /usr/local/bin/besu/bin/besu --data-path=/var/lib/besu --data-storage-format=BONSAI --sync-mode=SNAP storage x-trie-log prune
+```
+
+### Database not detected
+
+`java.lang.IllegalArgumentException: Trying to retain more trie logs than chain length (0), skipping pruning`
+
+Ensure you specify the correct `--data-path` for your node. 
+Your command should look similar to the following:
+
+```bash
+sudo /usr/local/bin/besu/bin/besu --data-path=/var/lib/besu --data-storage-format=BONSAI --sync-mode=SNAP storage x-trie-log prune
+```
+
+### Cannot store generated private key
+
+`java.lang.IllegalArgumentException: Cannot store generated private key`
+
+Ensure you specify the correct `--data-path` for your node. 
+Your command should look similar to the following:
+
+```bash
+sudo /usr/local/bin/besu/bin/besu --data-path=/var/lib/besu --data-storage-format=BONSAI --sync-mode=SNAP storage x-trie-log prune
+```
+
+### Valid keyPair not provided
```

Comment:
```suggestion
### Valid keypair not provided
```

Reply:
I am trying to mimic the error message. It is formatted as `keyPair`